src/keyring/KeyringController.ts

```diff
@@ -730,6 +741,114 @@ export class KeyringController extends BaseController<
     await this.#keyring.persistAllKeyrings();
     await this.fullUpdate();
   }
+
+  // Ledger Related methods
+
+  private async addLedgerKeyring(): Promise<LedgerKeyring> {
+    return await this.#keyring.addNewKeyring(KeyringTypes.ledger);
+  }
+
+  /**
+   * Retrieve the existing LedgerKeyring or create a new one.
+   *
+   * @returns The stored Ledger Keyring
+   */
+  async getLedgerKeyring(): Promise<LedgerKeyring> {
+    const keyring = this.#keyring.getKeyringsByType(KeyringTypes.ledger)[0];
+
+    if (keyring) {
+      return keyring;
+    }
+
+    return await this.addLedgerKeyring();
+  }
+
+  /**
+   * Connects to the ledger device by requesting some metadata from it.
+   *
+   * @param transport - The transport to use to connect to the device
+   * @param deviceId - The device ID to connect to
+   * @returns The name of the currently open application on the device
+   */
+  async connectLedgerHardware(
+    transport: Transport,
+    deviceId: string,
+  ): Promise<string> {
+    const keyring = await this.getLedgerKeyring();
+    keyring.setTransport(transport, deviceId);
+
+    const { appName } = await keyring.getAppAndVersion();
+
+    return appName;
+  }
+
+  /**
+   * Retrieve the first account from the Ledger device.
+   *
+   * @returns The default (first) account on the device
+   */
+  async unlockLedgerDefaultAccount(): Promise<{
```

Comment:
Nit: Maybe this would read better if named `unlockDefaultLedgerAccount`?

Reply:
[967d76b](https://github.com/MetaMask/controllers/pull/862/commits/967d76b01da77b7707a270eea022562b057389d4)